templates/code-style-validation.yml

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+parameters:
+  flake8_exclude: ""
+
+jobs:
+# Spin up a Linux VM for validating code.
+- job: code_style_validation
+  displayName: Code Style Validation
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+
+  # Use Python 3 for validating the code.
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.x'
+
+  # Install all the Python code validation packages.
+  - bash: python -m pip install black flake8
+    displayName: Install validators
+
+  # Ensure the code that was pushed has the proper formatting, except in the
+  # ui folder, which is auto-generated.
+  - bash: black --check . --exclude="ui\/.*py$"
```

Comment:
I think at some point it might be better to `pip install pre-commit` and run `pre-commit run -all` on the repo instead, which would catch all other error types. Plus, each repo will already have their own exclusion list properly set up in the pre-commit configuration file, so we won't have to try and be clever here.